### PR TITLE
chore: Block Signing protobufs 

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes https://github.com/hashgraph/hedera-services/issues/16422

- Adds protobufs and cnfigurations for Block Signing
- Adds empty `TssShareSignatureTransactionHandler` which will be implemented in next PR

### CODE REVIEW

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ */
+
+option java_package = "com.hedera.hapi.services.auxiliary.tss.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.services.auxiliary.tss">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The body of a transaction that contains a share signature produced by a node in a TSS network.
+ * This transaction will be gossipped with highest priority.
+ */
```
Comment (thomas-swirlds-labs) [inline hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:38]: nit: Every TransactionBody should now have a heading section for `### Block Stream Effects `

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.services.auxiliary.tss.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.services.auxiliary.tss">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The body of a transaction that contains a share signature produced by a node in a TSS network.
```
Comment (thomas-swirlds-labs) [inline hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:36]: nit: Limit and wrap column widths after 80 chars

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TssConfig.java [hunk 3]
```diff
@@ -23,10 +23,19 @@
...
- * @param timesToTrySubmission The number of times to retry a submission on getting an {@link IllegalStateException}
- * @param distinctTxnIdsToTry The number of distinct transaction IDs to try in the event of a duplicate id
- * @param keyActiveRoster A test-only configuration; set this to true to enable the process that will key the candidate roster with TSS key material.
+ *
+ * @param maxSharesPerNode                The maximum number of shares that can be assigned to a node.
+ * @param timesToTrySubmission            The number of times to retry a submission on getting an {@link IllegalStateException}
+ * @param retryDelay                      The delay between retries
+ * @param distinctTxnIdsToTry             The number of distinct transaction IDs to try in the event of a duplicate id
+ * @param keyCandidateRoster              A feature flag for TSS; set this to true to enable the process that will key
+ * @param keyActiveRoster                 A test-only configuration; set this to true to enable the process that will key the candidate
```
Comment (thomas-swirlds-labs) [inline hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TssConfig.java:32]: small typo - "that will key the **active** roster" ?

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TssConfig.java [hunk 4]
```diff
@@ -23,10 +23,19 @@
...
+ *                                        roster with TSS key material.
+ * @param enableLedgerId                  A feature flag for TSS; set this to true to enable the process that will key
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TssConfig.java:34]: yeah agreed. Both are not needed. Removed

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:36]: Done. Thanks Thomas

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:43]: Done. Thanks Thomas.

**Review by thomas-swirlds-labs — DISMISSED** [review]
LGTM. Thanks 👍  @Neeharika-Sompalli

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  uint64 share_index = 2;
+
+  /**
+   * The hash of the message that was signed by the share.
+   */
+  bytes message_hash = 3;
+
+  /**
+   * The share signature produced by the share.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:62]: This needs to be expanded as well.

**Review by jsync-swirlds — COMMENTED** [review]
The proto files need more detail, in appropriate format.

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ */
+message TssShareSignatureTransactionBody {
+  /**
+   * The hash of the roster containing the node whose share produced the
+   * share signature.
+   */
+  bytes roster_hash = 1;
+
+  /**
+   * The index of the share that produced this share signature.
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:59]: Changed this. Thanks @jsync-swirlds

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  uint64 share_index = 2;
+
+  /**
+   * The hash of the message that was signed by the share.
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:64]: Changed as per specification

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto (hunk 5 above)
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto]: Done as well. Thanks @jsync-swirlds

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+ */
+
+option java_package = "com.hedera.hapi.services.auxiliary.tss.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.services.auxiliary.tss">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The body of a transaction that includes a node's signature of a block hash using its private
+ * shares within the TSS process.
+ * This transaction will be prioritized for gossip transmission.
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:40]: There should be a summary line here, also, note that single newlines are collapsed in markdown, so you probably _intended_ to have a `<br/>` tag after `process.` at the end of the second line.

A _possible_ summary line might be `A TSS transaction Body.<br/>`.

Alternatively, this third line looks like a requirement:
```suggestion
 * A TSS transaction Body.<br/>
 * This transaction body communicates a node's signature of a block hash
 * using its private share within the TSS process.
 *
 * This transaction MUST be prioritized for low latency gossip transmission.
```

**Review by jsync-swirlds — COMMENTED** [review]
Good improvement, still a couple small items.

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * ### Block Stream Effects
+ * This transaction body will be present in the block stream. This will not have
+ * any state changes or transaction output or transaction result.
+ */
+message TssShareSignatureTransactionBody {
+  /**
+   * A SHA2-384 hash identifying the roster that includes the node whose share
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto:48]: I see. Just felt this one lie gives the insight. Will change as per the specification we are following.

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto (hunk 8 above)
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/auxiliary/tss/tss_share_signature.proto]: Fixed as per suggestion. Thanks @jsync-swirlds

**Review by tinker-michaelj — APPROVED** [review]
Very clean changeset, tyvm @Neeharika-Sompalli !

**Pull request merged**